plugins/inputs/prometheus/target_allocator.go

```diff
@@ -0,0 +1,192 @@
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/yaml.v3"
+)
+
+const DEFAULT_TLS_CA_FILE_PATH = "/etc/amazon-cloudwatch-observability-agent-cert/tls-ca.crt"
+
+type TargetAllocatorManager struct {
+	enabled             bool
+	manager             *tamanager.Manager
+	config              *otelpromreceiver.Config
+	host                component.Host
+	sm                  *scrape.Manager
+	dm                  *discovery.Manager
+	smLiveCh            chan struct{}
+	dmLiveCh            chan struct{}
+	taReadyCh           chan struct{}
+	reloadConfigHandler func(config *promconfig.Config)
+}
+
+func isPodNameAvailable() bool {
+	podName := os.Getenv(envconfig.PodName)
+	if podName == "" {
+		return false
+	}
+	return true
+}
+func loadConfigFromFilename(filename string) (*otelpromreceiver.Config, error) {
+	yamlFile, err := os.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	var stringMap map[string]interface{}
+	err = yaml.Unmarshal(yamlFile, &stringMap)
+	if err != nil {
+		return nil, err
+	}
+	componentParser := confmap.NewFromStringMap(stringMap)
+	if componentParser == nil {
+		return nil, fmt.Errorf("unable to parse config from filename %s", filename)
+	}
+	var cfg otelpromreceiver.Config
+	err = componentParser.Unmarshal(&cfg)
+	if err != nil {
+		return nil, err
+	}
+	return &cfg, nil
+}
+
+// Adapter from go-kit/log to zap.Logger
+func createLogger(level zapcore.Level) *zap.Logger {
+	// Create a base zap logger (you can customize it as needed)
+	zapCore := zapcore.NewCore(
+		zapcore.NewJSONEncoder(zap.NewProductionEncoderConfig()), // Use JSON encoder for zap
+		zapcore.AddSync(os.Stdout),                               // Output to stdout
+		level,                                                    // Set log level to Debug
+	)
+	// Create the zap logger
+	zapLogger := zap.New(zapCore)
+	return zapLogger
+}
+
+func createTargetAllocatorManager(filename string, logger log.Logger, sm *scrape.Manager, dm *discovery.Manager, smDoneCh, dmDoneCh chan struct{}) *TargetAllocatorManager {
+	tam := TargetAllocatorManager{
+		enabled:             false,
+		manager:             nil,
+		config:              nil,
+		host:                nil,
+		sm:                  sm,
+		dm:                  dm,
+		smLiveCh:            smDoneCh,
+		dmLiveCh:            dmDoneCh,
+		taReadyCh:           make(chan struct{}, 1),
+		reloadConfigHandler: nil,
+	}
+	err := tam.loadConfig(filename)
+	if err != nil {
+		level.Warn(logger).Log("msg", "Could not load config for target allocator from file", "filename", filename, "err", err)
+		return &tam
+	}
+	if tam.config == nil {
+		return &tam
+	}
+	tam.enabled = (tam.config.TargetAllocator != nil) && isPodNameAvailable()
+	if tam.enabled {
+		tam.loadManager()
+	}
+	return &tam
+}
+func (tam *TargetAllocatorManager) loadManager() {
```

Comment:
nit: function spacing
```suggestion
}

func (tam *TargetAllocatorManager) loadManager() {
```